### PR TITLE
create missing rules for cppcheck 1.82

### PR DESCRIPTION
**Pull request by ivangalkin**
This change is [Reviewable](https://reviewable.io/reviews/sonaropencommunity/sonar-cxx/1443)

### CODE REVIEW

File: cxx-sensors/src/main/resources/cppcheck.xml [hunk 1]
```diff
@@ -6345,7 +6345,1017 @@ The expression 'strcmp(x,"def") != 0' is suspicious. It overlaps 'strcmp(x,"abc"
     <remediationFunction>LINEAR</remediationFunction>
     <remediationFunctionGapMultiplier>1min</remediationFunctionGapMultiplier>
   </rule>
-  <!-- ########### add missing rules which seems not to be part of cppcheck errorlist ########### -->
+  <!-- ########### keys matching the schema <some-function-id>Called are not listed by the simple cppcheck -errorlist          ########### -->
+  <!-- ########### they become visible only if corresponding library was loaded, e.g. cppcheck -errorlist -library=posix.cfg   ########### -->
+  <!-- ########### see generate_cppcheck_resources.sh for more details                                                         ########### -->
+  <rule>
```
Comment (ivangalkin) [inline cxx-sensors/src/main/resources/cppcheck.xml]: similar to all existing obsolete/ not thread-safe functions the new ones have

```XML
<severity>MINOR</severity>
<type>CODE_SMELL</type>
```

Not sure it's really that harmless, but at least it's consistent

File: cxx-sensors/src/tools/cppcheck_createrules.py [hunk 2]
```diff
@@ -26,10 +26,14 @@
 import sys
 import xml.etree.ElementTree as et
 
-severity_2_priority = {
-    "error": "MAJOR",
-    "style": "MINOR"
-    }
+severity_2_sonarqube = {
```
Comment (ivangalkin) [inline cxx-sensors/src/tools/cppcheck_createrules.py]: not sure about this mapping too, but for the new rules we have to double-check the type/priority manually in any way

File: cxx-sensors/src/tools/generate_cppcheck_resources.sh [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+# these configurations contain additional rules about undesired (obsolete, not thread-safe etc) functions
+#
+# typical location:
+# * Red Hat based linux: /usr/share/cppcheck/
+# * cppcheck sources:    cppcheck/cfg/
+CFG_DIR=/usr/share/cppcheck/
+
+CPPCHECK_LIBRARY_ARGS=""
+for CFG in ${CFG_DIR}*.cfg; do
+   [ -e "${CFG}" ] && CPPCHECK_LIBRARY_ARGS="${CPPCHECK_LIBRARY_ARGS} --library=${CFG}" || echo "no *.cfg files exist in ${CFG_DIR}; no library specific rules will be generated!"
```
Comment (ivangalkin) [inline cxx-sensors/src/tools/generate_cppcheck_resources.sh]: for 1.82 the final argument list looks something like that:

cppcheck --errorlist --library=avr.cfg --library=bsd.cfg --library=cppcheck-lib.cfg --library=embedded_sql.cfg --library=gnu.cfg --library=gtk.cfg --library=microsoft_sal.cfg --library=motif.cfg --library=posix.cfg --library=qt.cfg --library=sdl.cfg --library=sfml.cfg --library=std.cfg --library=windows.cfg --library=wxwidgets.cfg

**Comment by guwirth** [comment]
@ivangalkin thanks for providing this. Can you rebase it please to see the changes better.

**Comment by ivangalkin** [comment]
@guwirth I mis-configured my git repository and will provide the new pull request instead of rebasing this one. Please see https://github.com/SonarOpenCommunity/sonar-cxx/pull/1444

**Pull request closed without merging**